### PR TITLE
[#1058] PublishingGuide: Provide a way to use GitHub actions to auto-generate the dashboard

### LINKED ISSUE #1058: Provide a way to use GitHub actions to auto-generate the dashboard
Current: users need to use Travis to auto-generate the dashboard periodically
Suggested: explore if GitHub actions can be given as an alternative

### PR DESCRIPTION
**Pull request by niqiukun**
Fixes #1058 

To be reviewed together with PR https://github.com/reposense/publish-RepoSense/pull/7

```
Currently, the only way to leverage on CI and auto-generate the dashboard
is by using Travis CI, a third-party service.

We can provide a way to use GitHub Actions to auto-generate the dashboard
so that users have an alternative option now and do not have to rely on 
third-party services for deployment.

Let's add the documentation which shows users how to auto-generate the 
dashboard using GitHub Actions.
```

### CODE REVIEW

**Comment by damithc** [comment]
How's this PR going?

File: docs/PublishingGuide.md [hunk 1]
```diff
@@ -22,6 +24,21 @@
 It takes a few minutes for report generation. Meanwhile, you can monitor the progress live at [Travis-CI's Builds](https://travis-ci.org/dashboard/builds). <br/>
 Try accessing your site again when a green tick appears beside your fork.
 
+### Using GitHub Actions
+
+1. Fork publish-RepoSense repository using this [link](https://github.com/RepoSense/publish-RepoSense/fork)
+1. Follow this [section](#granting-write-access-to-reposense-for-publishing) to generate a `personal access token` or `deploy key` on GitHub for report publishing 
+1. Go to the [secrets settings](../../../../publish-RepoSense/settings/secrets) of your publish-RepoSense fork, add a new secret as `ACCESS_TOKEN` or `DEPLOY_KEY` depending on your earlier choice and paste the content that was copied earlier to its value field; then click **Add secret** <br/>
```
Comment (jamessspanggg) [inline docs/PublishingGuide.md]: Can provide a screenshot of the secrets settings, similar to the screenshot of the travis environment variables settings

File: docs/PublishingGuide.md [hunk 2]
```diff
@@ -22,6 +24,22 @@
...
+![GitHub Actions Secrets](images/publishingguide-secrets.png "GitHub Actions Secrets")
+1. Edit [run.sh](../../../../publish-RepoSense/edit/master/run.sh), [repo-config.csv](../../../../publish-RepoSense/edit/master/configs/repo-config.csv) and [author-config.csv](../../../../publish-RepoSense/edit/master/configs/author-config.csv) to customize the command line parameters or repositories to be analyzed <br/>
+*Read our [User Guide](UserGuide.md#customizing-the-analysis) for more information*
```
Comment (Tejas2805) [inline docs/PublishingGuide.md]: Add this line in a different manner. The rendered view file doesn't look good. Maybe can just add it to the previous line or use `>`.

Comment (niqiukun) [reply docs/PublishingGuide.md]: This line was copied from `Using Travis` section. I did not make changes to it for consistency.

**Comment by niqiukun** [comment]
> There are missing punctuation marks. Please do add them, otherwise, the sentences look a bit weird. Do ask for a review again, when done.

Could you highlight what punctuation marks you are refering to?

File: docs/PublishingGuide.md (hunk 2 above)
Comment (Tejas2805) [reply docs/PublishingGuide.md]: Yes noticed that. Can you just edit in both the places? Looks a bit weird currently.

Screen Shot 2020-03-08 at 10 31 10 PM

> Could you highlight what punctuation marks you are referring to?

Add full stops at the end of the steps and sentences. You can do this for both the sections too.

File: docs/PublishingGuide.md [hunk 3]
```diff
@@ -53,6 +73,14 @@ Alternatively, you can manually trigger an update.
 1. Go to [your fork in Travis-CI](https://travis-ci.org/search/publish-RepoSense/), click on **More options** on the right then **Trigger build**
 1. In the pop up, click **Trigger custom build**
 
+#### Using GitHub Actions
+
+1. Edit [main.yml](../../../../publish-RepoSense/edit/master/.github/workflows/main.yml) and uncomment the `schedule:` section
+1. You may change the expression after `cron:` to a schedule of your choice; read more about cron syntax [here](https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule)
+1. Commit your changes
+
+Alternatively, you can trigger an update by pushing an empty commit to your fork. GitHub Actions currently do not support manual triggers.
```
Comment (0blivious) [inline docs/PublishingGuide.md]: Maybe change to `Currently, the GitHub Actions UI doesn't support the manual execution of workflows`

**Comment by fzdy1914** [comment]
Can resolve the conflicts please?

**Comment by niqiukun** [comment]
> Can resolve the conflicts please?

Resolved.

**Pull request merged**